Tests/Common/Util/StringAnyEncodable.swift

```diff
@@ -85,14 +87,16 @@ class StringAnyEncodableTest: UnitTest {
     }
 
     func test_stringanyencodable_encodes_complex_data() {
-        let expect = DummyData(boolean: true, numeric: 1, testValue: "foo", dict: ["test": "value"], array: [1, 2, 4])
+        let expect = DummyData(boolean: true, numeric: 1, testValue: "foo", dict: ["test": "value"], array: [1, 2, 4], hashables: ["color": ["Red", "Green", "Blue"]])
 
+        // React native wrap some values in AnyHashable
         let data = [
             "testValue": "foo",
             "numeric": 1,
             "boolean": true,
             "array": [1, 2, 4],
-            "dict": ["test": "value"] as [String: Any]
+            "dict": ["test": "value"] as [String: Any],
+            "hashables": [AnyHashable("color"): [AnyHashable("Red"), AnyHashable("Green"), AnyHashable("Blue")]] as [AnyHashable: [AnyHashable]]
```

Comment:
curious why do we need to provide type `AnyHashable` here? 

maybe add another test if you want to check for `AnyHashable`, but for this we should check we are expect and getting the same map?

Reply:
Sorry for not making it that clear earlier, But I tried the following code on both react native and Flutter.

```
CustomerIO.setProfileAttributes(attributes: {
  'dietary_restrictions-1': 'VE',
  'dietary_restrictions-2': JSON.stringify(['VE', 'P']),
  'dietary_restrictions-3': ['VE', 'P'],
});
```

And both of these wrappers had similar output as follows:

```
["attributes": AnyHashable([AnyHashable("dietary_restrictions-3"): AnyHashable([AnyHashable("VE"), AnyHashable("P")]), AnyHashable("dietary_restrictions-2"): AnyHashable("[VE, P]"), AnyHashable("dietary_restrictions-1"): AnyHashable("VE")])]
```

So looks like both of these wrappers pass data wrapped in `AnyHashable`. This might be for some flexibility, but I'm not really sure about the reason. Couldn't even find any relevant link too but this was just result of running and debugging different values from both of these wrappers. 🤐 

Adding separate test function and DummyData class for wrapper might make it more clear? Let me know if you folks have more ideas on it.

Reply:
With new log message, the output for case 3 is

```
Tried to convert ["dietary_restrictions-3": AnyHashable([AnyHashable("VE"), AnyHashable("P")])] into [String: AnyEncodable] but the data type is not Encodable.
```

Reply:
Updated test cases which might help addressing this

Reply:
The latest change that you made to this test function we are discussing looks like we have resolved this. 

I made a [comment](https://github.com/customerio/customerio-ios/pull/358#discussion_r1273529294) about how I believe using `AnyHashable(X)` anywhere in our test class doesn't add value. That can be discussed there. I'll resolve this since it looks like Shahroz and I's comment has been addressed.